### PR TITLE
feature(SPV-1106): introduce new bhs errors

### PR DESCRIPTION
**Pull request by dzolt-4chain**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/block-headers-service/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/block-headers-service/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

**Review by chris-4chain — APPROVED** [review]
LGTM 🙈 😛

**Review by jakubmkowalski — APPROVED** [review]

File: service/chain_service.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func (cs *chainService) createHeader(hash *domains.BlockHash, bs *domains.BlockH
 
 func (cs *chainService) previousHeader(bs *domains.BlockHeaderSource) (*domains.BlockHeader, error) {
 	h, err := cs.Repositories.Headers.GetHeaderByHash(bs.PrevBlock.String())
-	if h == nil && err != nil && err.Error() == "could not find hash" {
+	if h == nil && err != nil && err.Error() == "Header not found" {
```
Comment (chris-4chain) [inline service/chain_service.go:189]: Can't it be done using `errors.Is` ?

I'm not a fun of comparing error messages especially when it is a part of some logic/algorithm.

File: bhserrors/definitions.go
Comment (jakubmkowalski) [inline bhserrors/definitions.go:1]: In spv-wallet messages start with a lowercase letter.  Here sometimes it is lowercase sometimes uppercase. It is best to choose one method and use it consistently. In my opinion, it is better to follow the convention from spv-wallet.

File: database/sql/tokens.go [hunk 2]
```diff
@@ -39,19 +38,22 @@ func (h *HeadersDb) CreateToken(ctx context.Context, token *dto.DbToken) error {
...
 	if _, err := tx.NamedExecContext(ctx, h.db.Rebind(sqlInsertToken), *token); err != nil {
-		return errors.Wrap(err, "failed to insert token")
+		return bhserrors.ErrCreateToken.Wrap(err)
 	}
-	return errors.Wrap(tx.Commit(), "failed to commit tx")
+
+	err = tx.Commit()
+	if err != nil {
+		return bhserrors.ErrCreateToken.Wrap(err)
+	}
```
Comment (jakubmkowalski) [inline database/sql/tokens.go:47]: optional suggestion
```
if err := tx.Commit(); err != nil { 
   return bhserrors.ErrCreateToken.Wrap(err) 
}
```

File: database/sql/tokens.go [hunk 3]
```diff
@@ -67,8 +69,14 @@ func (h *HeadersDb) DeleteToken(ctx context.Context, token string) error {
...
 	if _, err = tx.NamedExecContext(ctx, h.db.Rebind(sqlDeleteToken), map[string]interface{}{"token": token}); err != nil {
-		return errors.Wrap(err, "failed to delete token")
+		return bhserrors.ErrDeleteToken.Wrap(err)
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return bhserrors.ErrDeleteToken.Wrap(err)
+
 	}
```
Comment (jakubmkowalski) [inline database/sql/tokens.go:79]: optional suggestion
```
if err := tx.Commit(); err != nil { 
   return bhserrors.ErrDeleteToken.Wrap(err)
}
```

File: database/sql/webhooks.go [hunk 4]
```diff
@@ -50,29 +50,34 @@ func (h *HeadersDb) CreateWebhook(ctx context.Context, rWebhook *dto.DbWebhook)
...
+		return bhserrors.ErrCreateWebhook.Wrap(err)
 	}
-	return errors.Wrap(tx.Commit(), "failed to commit tx")
+
+	err = tx.Commit()
+	if err != nil {
+		return bhserrors.ErrCreateWebhook.Wrap(err)
+	}
+
+	return nil
```
Comment (jakubmkowalski) [inline database/sql/webhooks.go:61]: optional suggestion
```
if err := tx.Commit(); err != nil { 
   return bhserrors.ErrCreateWebhook.Wrap(err)
}
```

File: database/sql/webhooks.go [hunk 5]
```diff
@@ -89,10 +94,15 @@ func (h *HeadersDb) DeleteWebhookByURL(ctx context.Context, url string) error {
...
+		return bhserrors.ErrDeleteWebhook.Wrap(err)
 	}
 
-	return errors.Wrap(tx.Commit(), "failed to commit tx")
+	err = tx.Commit()
+	if err != nil {
+		return bhserrors.ErrDeleteWebhook.Wrap(err)
+	}
+
+	return nil
```
Comment (jakubmkowalski) [inline database/sql/webhooks.go:105]: optional suggestion
```
if err := tx.Commit(); err != nil { 
   return bhserrors.ErrDeleteWebhook.Wrap(err)
}
```

File: transports/http/endpoints/api/headers/header_endpoints_test.go [hunk 6]
```diff
@@ -37,19 +37,19 @@ func TestGetHeaderByHash(t *testing.T) {
...
 		}
 
 		// when
 		res := bhs.API().Call(getHeaderByHash("123"))
 
 		// then
 		assert.Equal(t, res.Code, expectedResult.code)
-		body, _ := io.ReadAll(res.Body)
-		assert.EqualBytes(t, body, expectedResult.body)
+		// exp, _ := json.MarshalIndent(.body, "", "  ")
```
Comment (jakubmkowalski) [inline transports/http/endpoints/api/headers/header_endpoints_test.go:51]: Commented out, delete if not needed

File: database/sql/webhooks.go [hunk 7]
```diff
@@ -89,10 +94,15 @@ func (h *HeadersDb) DeleteWebhookByURL(ctx context.Context, url string) error {
...
 	if _, err = tx.NamedExecContext(ctx, h.db.Rebind(sqlDeleteWebhookByURL), params); err != nil {
-		return errors.Wrap(err, "failed to delete webhook")
+		return bhserrors.ErrDeleteWebhook.Wrap(err)
 	}
 
-	return errors.Wrap(tx.Commit(), "failed to commit tx")
+	err = tx.Commit()
+	if err != nil {
+		return bhserrors.ErrDeleteWebhook.Wrap(err)
+	}
```
Comment (jakubmkowalski) [inline database/sql/webhooks.go:103]: optional suggestion
```
if err := tx.Commit(); err != nil { 
   return bhserrors.ErrDeleteWebhook.Wrap(err)
}
```

File: service/chain_service.go (hunk 1 above)
Comment (dzolt-4chain) [reply service/chain_service.go:189]: It can, sure let me change it out

File: bhserrors/definitions.go
Comment (dzolt-4chain) [reply bhserrors/definitions.go:1]: Right, changed to lowercase

File: database/sql/tokens.go (hunk 2 above)
Comment (dzolt-4chain) [reply database/sql/tokens.go:46]: simplified

File: database/sql/tokens.go (hunk 3 above)
Comment (dzolt-4chain) [reply database/sql/tokens.go:77]: simplified

File: database/sql/webhooks.go (hunk 4 above)
Comment (dzolt-4chain) [reply database/sql/webhooks.go:60]: simplified

File: database/sql/webhooks.go (hunk 7 above)
Comment (dzolt-4chain) [reply database/sql/webhooks.go:101]: simplified

File: transports/http/endpoints/api/headers/header_endpoints_test.go (hunk 6 above)
Comment (dzolt-4chain) [reply transports/http/endpoints/api/headers/header_endpoints_test.go]: ah ty for noticing, removed

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**